cudamapper/include/claraparabricks/genomeworks/cudamapper/index.hpp

```diff
@@ -215,19 +250,34 @@ class IndexHostCopyBase
     /// \return window_size_
     virtual std::uint64_t window_size() const = 0;
 
-    /// \brief Constructor
-    /// \brief creates a copy of input processed index on the host
+    /// \brief Starts creating a copy of index on the host
+    /// Copy is done asynchronously and one should wait for it to finish with finish_copying()
     /// \param index - pointer to computed index parameters (vectors of sketch elements) on GPU
     /// \param first_read_id - representing smallest read_id in index
     /// \param kmer_size - number of basepairs in a k-mer
     /// \param window_size the number of adjacent k-mers in a window, adjacent = shifted by one basepair
     /// \param cuda_stream D2H copy is done on this stream
     /// \return - an instance of IndexHostCopyBase
-    static std::unique_ptr<IndexHostCopyBase> create_cache(const Index& index,
-                                                           const read_id_t first_read_id,
-                                                           const std::uint64_t kmer_size,
-                                                           const std::uint64_t window_size,
-                                                           const cudaStream_t cuda_stream = 0);
+    static std::unique_ptr<IndexHostCopyBase> create_host_copy(const Index& index,
```

Comment:
Would it make sense to call this `create_host_copy_async` to emphasize that you have to wait afterwards?

Reply:
I agree. We should generally pay more attention to signaling which methods are async and which are not and this is a good first step.
done